New and improved F::load*() methods, prevent access to internal resources by included files

lukasbestle opened the pull request:
## Describe the PR

<!-- A clear and concise description of the bug the PR fixes or the feature the PR introduces. -->

- New `F::loadOnce()` method for safely including PHP files using `include_once()`
- `F::load()` now supports passing variables to the included file with a new `$data` argument
- Files included with `F::load()` can no longer overwrite the provided `$fallback` variable
- PHP files included by Kirby (e.g. from plugins, models, controllers etc.) can no longer access `protected` methods of Kirby classes

**Note:** The fourth change is a breaking-change.

## Related issues

<!-- PR relates to issues in the `kirby` or `idea` repo: -->

- Fixes #2394

## Ready?

<!-- If you feel like you can help to check off the following tasks, that'd be great. If not, don't worry - we will take care of it. -->

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Fixed code style issues with CS fixer and `composer fix`
- [x] Added in-code documentation (if needed)

bastianallgeier reviewed (commented):
@lukasbestle I like `F::loadOnce`, but `F::loadNative` sounds unintuitive to me. What about `F::loadScoped` or `F::loadIsolated` or another argument for `F::load`?

lukasbestle committed:
F::load(): Prevent overwriting $fallback #2394

lukasbestle committed:
F::load(): Support for data variables #2394

lukasbestle committed:
New F::loadOnce() method #2394

lukasbestle committed:
Switch from include/require to F::load*() #2394

This ensures that the included file can neither access any local variables from the calling methods nor protected methods from the calling classes.

See https://www.php.net/manual/en/function.include.php#example-124.

bastianallgeier committed:
Merge branch 'develop' into fix/2394-safe-plugin-include
